src/panel/chart.tsx

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import { PureComponent } from 'react';
+import { ResultTableStore } from './resultTableStore';
+import * as d3 from 'd3';
+
+export interface ChartProps<G> {
+    store: ResultTableStore<G>;
+}
+@observer export class Chart<G> extends PureComponent<ChartProps<G>> {
+    private ref!: SVGSVGElement;
```

Comment:
Why is there an exclamation here?

Reply:
tells the TypeScript compiler that a value typed as optional cannot be null or undefined